src/services/services.ts

```diff
@@ -4316,6 +4361,159 @@ namespace ts {
                     }
                 }
             }
+
+            function getStringLiteralCompletionEntriesFromModuleNames(node: StringLiteral) {
+                const literalValue = node.text;
+                let result: CompletionEntry[];
+
+                const isRelativePath = startsWith(literalValue, ".");
+                const scriptDir = getDirectoryPath(node.getSourceFile().path);
+                if (isRelativePath || isRootedDiskPath(literalValue)) {
+                    result = getCompletionEntriesForDirectoryFragment(literalValue, scriptDir, getSupportedExtensions(program.getCompilerOptions()), /*includeExtensions*/false);
+                }
+                else {
+                    // Check for node modules
+                    result = getCompletionEntriesForNonRelativeModules(literalValue, scriptDir);
+                }
+
+                return {
+                    isMemberCompletion: false,
+                    isNewIdentifierLocation: false,
+                    entries: result
+                };
+            }
+
+            function getCompletionEntriesForDirectoryFragment(fragment: string, scriptPath: string, extensions: string[], includeExtensions: boolean): CompletionEntry[] {
+                // Complete the path by looking for source files and directories
+                const result: CompletionEntry[] = [];
+
+                const toComplete = getBaseFileName(fragment);
+                const absolutePath = normalizeSlashes(host.resolvePath(isRootedDiskPath(fragment) ? fragment : combinePaths(scriptPath, fragment)));
+                const baseDir = toComplete ? getDirectoryPath(absolutePath) : absolutePath;
+
+
+                if (directoryProbablyExists(baseDir, host)) {
+                    // Enumerate the available files
+                    const files = host.readDirectory(baseDir, extensions, /*exclude*/undefined, /*include*/["./*"]);
+                    forEach(files, (f) => {
+                        const fName = includeExtensions ? getBaseFileName(f) : removeFileExtension(getBaseFileName(f));
+
+                        if (startsWith(fName, toComplete)) {
+                            result.push({
+                                name: fName,
+                                kind: ScriptElementKind.unknown,
+                                kindModifiers: ScriptElementKindModifier.none,
+                                sortText: fName

```

Comment:
we will need to add a new concept of completion span (similar to Roslyn's, see: http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionItem.cs,39).

The assumption now is the at the caller will compute the span as the first non-identifier character from the completion location. this is not true with file path, so we should provide a span to the caller.

the span should be the span of the text in the literal, moduluo the quotes. e.g. `".\mod"` => span of `.\mod`